plugins/modules/system/lvol.py

```diff
@@ -553,25 +556,28 @@ def main():
 
         else:
             # resize LV based on absolute values
-            tool = None
+            cmd = None
             if float(size) > this_lv['size'] or size_operator == '+':
                 tool = module.get_bin_path("lvextend", required=True)
+                cmd = tool
             elif shrink and float(size) < this_lv['size'] or size_operator == '-':
                 if float(size) == 0:
                     module.fail_json(msg="Sorry, no shrinking of %s to 0 permitted." % (this_lv['name']))
                 if not force:
                     module.fail_json(msg="Sorry, no shrinking of %s without force=yes." % (this_lv['name']))
                 else:
                     tool = module.get_bin_path("lvreduce", required=True)
-                    tool = '%s %s' % (tool, '--force')
+                    cmd = '%s %s' % (tool, '--force')
 
-            if tool:
+            if cmd:
                 if resizefs:
-                    tool = '%s %s' % (tool, '--resizefs')
+                    cmd = '%s %s' % (cmd, '--resizefs')
                 if size_operator:
-                    cmd = "%s %s -%s %s%s%s %s/%s %s" % (tool, test_opt, size_opt, size_operator, size, size_unit, vg, this_lv['name'], pvs)
+                    cmd = "%s %s -%s %s%s%s %s/%s" % (cmd, test_opt, size_opt, size_operator, size, size_unit, vg, this_lv['name'])
                 else:
-                    cmd = "%s %s -%s %s%s %s/%s %s" % (tool, test_opt, size_opt, size, size_unit, vg, this_lv['name'], pvs)
+                    cmd = "%s %s -%s %s%s %s/%s" % (cmd, test_opt, size_opt, size, size_unit, vg, this_lv['name'])
+                if tool.endswith('lvextend'):
```

Comment:
Better use a boolean flag (set above) than string manipulation to detect this case.